boot/zephyr/nrf_cleanup.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <hal/nrf_rtc.h>
+#include <hal/nrf_uarte.h>
+#include <hal/nrf_clock.h>
+
+#if CONFIG_HAS_HW_NRF_RTC0 || CONFIG_HAS_HW_NRF_RTC1 || CONFIG_HAS_HW_NRF_RTC2
+static inline void nrf_cleanup_rtc(NRF_RTC_Type * rtc_reg)
+{
+    uint32_t mask = NRF_RTC_INT_TICK_MASK     |
+                    NRF_RTC_INT_OVERFLOW_MASK |
+                    NRF_RTC_INT_COMPARE0_MASK |
+                    NRF_RTC_INT_COMPARE1_MASK |
+                    NRF_RTC_INT_COMPARE2_MASK |
+                    NRF_RTC_INT_COMPARE3_MASK;
+
+    nrf_rtc_task_trigger(rtc_reg, NRF_RTC_TASK_STOP);
+    nrf_rtc_event_disable(rtc_reg, mask);
+    nrf_rtc_int_disable(rtc_reg, mask);
+}
+#endif
+
+void nrf_cleanup_clock(void)
+{
+    //nrfx_clock_stop(NRF_CLOCK_DOMAIN_LFCLK);
+    //nrfx_clock_stop(NRF_CLOCK_DOMAIN_HFCLK);
+    //nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_LFCLKSTOP);
+    
+#if NRF_CLOCK_HAS_HFCLK192M
+    //nrfx_clock_stop(NRF_CLOCK_DOMAIN_HFCLK192M);
+#endif
+#if NRF_CLOCK_HAS_HFCLKAUDIO
+    //nrfx_clock_stop(NRF_CLOCK_DOMAIN_HFCLKAUDIO);
```

Comment:
@ktaborowski let me know whether you are need this commented-out line?

Reply:
Our use case seems to work with these nrfx_colock_stop lines commented out